torch/csrc/jit/script/sugared_value.h

```diff
@@ -64,6 +66,12 @@ struct TORCH_API SugaredValue
     throw ErrorReport(loc) << kind() << " cannot be used as a type";
   }
 
+  virtual std::shared_ptr<SugaredValue> iter(
```

Comment:
can you add a comment for how this gets used and what is expected of the resulting object?